helix-term/src/application.rs

```diff
@@ -1029,6 +1029,61 @@ impl Application {
                             .collect();
                         Ok(json!(result))
                     }
+                    MethodCall::ShowDocument(params) => {
+                        use helix_view::editor::Action;
+
+                        if params.external.unwrap_or(false) {
+                            // TODO: Implement this
+                            Ok(json!(lsp::ShowDocumentResult { success: false }))
+                        } else {
+                            match params.uri.to_file_path() {
+                                Err(_) => {
+                                    let err = format!(
+                                        "unable to convert URI to filepath: {}",
+                                        params.uri
+                                    );
+                                    self.editor.set_error(err);
+                                    Ok(json!(lsp::ShowDocumentResult { success: false }))
+                                }
+                                Ok(path) => {
+                                    match self.editor.open(&path, Action::Replace) {
+                                        Err(err) => {
+                                            let err = format!(
+                                                "failed to open path: {:?}: {:?}",
+                                                params.uri, err
+                                            );
+                                            self.editor.set_error(err);
+                                            Ok(json!(lsp::ShowDocumentResult { success: false }))
+                                        }
+                                        Ok(_) => {
+                                            if let Some(range) = params.selection {
+                                                let (view, doc) = current!(self.editor);
+                                                let definition_pos = range.start;
+                                                // TODO: convert inside server
+                                                if let Some(new_pos) = lsp_pos_to_pos(
+                                                    doc.text(),
+                                                    definition_pos,
+                                                    helix_lsp::OffsetEncoding::Utf8,
```

Comment:
Here we should use the `language_server.offset_encoding()`: [this block](https://github.com/helix-editor/helix/blob/8043959265366ed0c945b7f868337a3f9823a747/helix-term/src/application.rs#L1034-L1040) can be moved above the `match call { .. }` and then we should also use the `language_server.offset_encoding()` [here](https://github.com/helix-editor/helix/blob/8043959265366ed0c945b7f868337a3f9823a747/helix-term/src/application.rs#L986-L990)

Reply:
I had to add a `get_by_id_arc` methods to `Registry` as the original `get_by_id` returned a immutable reference, meaning that `self.editor` could not be borrowed mutably.